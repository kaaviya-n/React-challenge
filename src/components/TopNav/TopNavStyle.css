.topNavWrapper {
    color: #ffffff;
    display: flex;
    justify-content: space-between;
    flex-grow: 1;
    padding: 15px;
    height: 3rem;
}
.arrowList {
    display: flex;
    align-items: center;
}
.arrowWrapper {
    padding: 12px;
    background-color: #000000;
    border-radius: 30px;
    font-size: 22px;
    margin-right: 22px;
}
.loginButton {
    padding: 12px 30px;
    border-radius: 25px;
    border-color: transparent;
    font-size: 20px;
    font-weight: 600;
}
.image {
    width: 10rem;
    height: 4rem;
  }