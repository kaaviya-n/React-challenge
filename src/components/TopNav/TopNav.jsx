import React from 'react';
import { useNavigate } from 'react-router-dom';
import PropTypes from 'prop-types';
import { MdArrowBackIosNew } from 'react-icons/md';

import SpotifyLogo from '../../assets/Spotify-Logo.svg';

import './TopNavStyle.css';

const TopNav = ({ goBack }) => {
	const navigate = useNavigate();
  
	const handleLogout = () => {
		localStorage.removeItem('token');
		navigate('/login');
	};

	return (
		<div className="topNavWrapper">
			<div className="arrowList">
				<MdArrowBackIosNew className="arrowWrapper" onClick={goBack} />
				<img className="image" src={SpotifyLogo} alt="Logo" />
			</div>    
			<button className="loginButton" onClick={handleLogout}>
          		Logout
			</button>
		</div>
	);
};

TopNav.propTypes = {
	goBack: PropTypes.func
};

export default TopNav;
