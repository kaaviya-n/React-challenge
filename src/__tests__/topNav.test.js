import React from 'react';
import { HashRouter } from 'react-router-dom';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import TopNav from '../components/TopNav/TopNav';

afterEach(cleanup);

const TopNavJSX = <TopNav goBack={() => {}} />;

it('UI testing - Top Navigation', () => {
	const tree = renderer.create(<HashRouter>{TopNavJSX}</HashRouter>).toJSON();
	expect(tree).toMatchSnapshot();
});

it('renders Top Nav component', () => {
	render(TopNavJSX, {wrapper: HashRouter});
});

it('Should render without throwing an error', () => {
	expect(render(
		TopNavJSX, {wrapper: HashRouter}
	)).not.toBeNull();
});

it(' Check for Logout text', () => {
	const { getByText } = render(
		TopNavJSX, {wrapper: HashRouter}
	);
	expect(getByText('Logout')).toBeInTheDocument();
});