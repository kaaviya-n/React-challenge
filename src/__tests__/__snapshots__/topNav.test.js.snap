// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI testing - Top Navigation 1`] = `
<div
  className="topNavWrapper"
>
  <div
    className="arrowList"
  >
    <svg
      className="arrowWrapper"
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M17.77 3.77L16 2 6 12l10 10 1.77-1.77L9.54 12z"
      />
    </svg>
    <img
      alt="Logo"
      className="image"
      src={
        Object {
          "blurDataURL": "data:image/png;base64,imagedata",
          "height": 24,
          "src": "/img.jpg",
          "width": 24,
        }
      }
    />
  </div>
  <button
    className="loginButton"
    onClick={[Function]}
  >
    Logout
  </button>
</div>
`;
